README.md

```diff
@@ -2,6 +2,128 @@
+            │          └── ...
+            └── v2/
+                 └── ...
+    ```
+
+-   **We use open source and open standards as much as possible.**
+    This means it's more likely tools will continue to exist that can understand the files in the archive, even if all of our code is no longer in use.
+
+    The unit of storage is "bags" in the [BagIt packaging format](https://tools.ietf.org/html/rfc8493).
+    BagIt is an open standard developed by the Library of Congress.
+
+
+
+## High-level design
+
+<figure>
+  <img src="high_level_design.svg">
+  <figcaption>
+    Icons:
+    <a href="https://thenounproject.com/term/paper-bag/28579/">Paper bag</a> by Dorian Dance. From The Noun Project, CC BY.
+    <a href="https://thenounproject.com/term/process/2473979/">Process</a>, by Alice Design. From The Noun Project, CC BY.
+    <a href="https://aws.amazon.com/architecture/icons/">Bucket</a>, from the AWS Architecture icons.
+  </figcaption>
+</figure>
+
+The user uploads a "bag" in the BagIt packaging format.
+
+The storage service unpacks the bag, and verifies the checksums in the BagIt manifest.
+If the checksums match the files, it creates three copies of the bag:
+
+*   One in Amazon S3: the warm copy, for day-to-day access
+*   One in Amazon Glacier: a backup copy
+*   One in Azure Blob Storage: a different cloud provider and geographic location, for extra redundancy (not implemented as of February 2020)
+
+It then re-verifies the three copies of the bag to ensure they were stored correctly, and reports the bag as successfully stored to the user.
+
+**The storage service stores exactly the bytes you give it; no more, no less.**
+It does not do any introspection of the bag contents, or change its behaviour based on the files a bag contains.
+
+
+
+## Identifiers
+
+For long-term portability, it's important to use human-readable and understandable identifiers (not [UUIDs](https://en.wikipedia.org/wiki/Universally_unique_identifier)).
+The storage service does not create its own identifiers for bags -- the user has to provide their own identifier.
+This allows bags to be matched to records in systems outside the storage service (for example, the library catalogue).
+
+Bags in the storage service have a three-part identifier:
+
+*   **Space:** the broad category of a bag.
+    Examples: `digitised`, `born-digital`.
+
+*   **External identifier:** the identifier of a bag within a space.
+    This is typically an identifier from another system, which matches this bag to that record.
+    Examples: `b31497652`, `PP/CRI/A/2`.
+
+*   **Version:** an auto-incrementing version assigned by the storage service.
+    This tracks distinct versions of a (space, external identifier) pair.
+    Examples: `v1`, `v2`, `v3`.
+
+These three parts can be combined into a single string, which uniquely identifies a bag; for example `digitised/b31497652/v2`.
+This identifier is also the path to the root of the bag inside our storage buckets.
+
```

Comment:
I'm a little unclear as to what parts of the ID are supplied vs calculated.

__Space__: Does the client define the space?
__Source ID__: Easy, supplied by client
 __Version__: Assuming this is controlled by the storage service?

Reply:
Added another line; make more sense now?

Reply:
Nothing to do with the docs, but the space can be anything?

Reply:
The space doesn't allow slashes (and if you try to store a bag while passing a space that contains a slash, it gets rejected). Everything else is okay.

Not allowing slashes in the space/version means that if you have an identifier like

```
born-digital/PP/CRI/A/2/v1
```

you can unambiguously decide which bit is the space, external identifier and version.

(If slashes were allowed, you wouldn't be able to distinguish between a space `born-digital`, `born-digital/PP` or even `born-digital/PP/CRI`.)

And it has to be a valid S3 key, which I think is all Unicode but I guess one day we might find out otherwise? 🤷‍♀

Reply:
See https://github.com/wellcomecollection/storage-service/blob/master/common/src/main/scala/uk/ac/wellcome/platform/archive/common/storage/models/StorageSpace.scala (which I'll explain in the developer docs)